invenio_users_resources/records/api.py

```diff
@@ -367,6 +397,49 @@ def get_record_by_name(cls, name):
                 return None
             return cls.from_model(role)
 
+    @classmethod
+    def create(cls, data):
+        """Create a new group/role and store it in the database."""
+        try:
+            # Validate before attempting DB write to avoid UOW commit failures
+            _validate_group_data(data)
+            # Create Role
+            role = current_datastore.create_role(**data)
+            return cls.from_model(role)
+        except ValidationError:
+            raise
+        except IntegrityError:
```

Comment:
I don't think you need this one here. You are already checking inside `_validate_group_data`. Am I missing something?